src/box/alter.cc

```diff
@@ -534,6 +534,12 @@ space_def_new_from_tuple(struct tuple *tuple, uint32_t errcode,
 			 "local space can't be synchronous");
 		return NULL;
 	}
+	if (!tuple_constraint_def_check_names(opts.constraint_def,
+					      opts.constraint_count)) {
+		diag_set(ClientError, ER_WRONG_SPACE_OPTIONS, "check "
+			 "constraints and foreign keys must have unique names");
+		return NULL;
+	}
```

Comment:
Better put it in `space_opts_decode`, I think, because that's where we create constraints.

Reply:
Done.